src/App.tsx

```diff
@@ -1,22 +1,32 @@
-import { useState } from 'react'
+import { useState, useEffect } from 'react'
 import logo from './logo.svg'
 import './App.css'
 import { invoke } from '@tauri-apps/api'
 
-function App() {
+interface BaseParams {
+  setState: (state: number) => void,
+}
+const AuthPage = (props: BaseParams) => {
+
+  const [authUrl, setAuthUrl] = useState("");
+
+  //We need to wrap this in useEffect so it only happens on first render, otherwise, on state change, react would rerender again and loop
+  useEffect(() => {
+    invoke('get_auth_url').then((url) => {
+      setAuthUrl(String(url));
+    })
+  }, []);
+
 
-  invoke('get_auth_url').then((url) => {
-    document.getElementById("auth_url")?.setAttribute("href", String(url));
-  })
   return (
-    <div className="App">
+    < div className="App">
```

Comment:
Probably this space character between `<` and `div` doesn't really harm anything -- but -- you might consider using a adding a linter to VSCode that will auto-format your code on each save so it always looks nice :)